ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -324,11 +327,18 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 			}
 		}
 	}
-	return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, nil
+
+	if isJobPendingOrRunning(rayJobInstance.Status.JobStatus) {
+		// Requeue the RayJob to poll its status from the running Ray job
+		return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, nil
+	}
+	// Otherwise only reconcile the RayJob upon new events for watched resources
+	// to avoid infinite reconciliation.
+	return ctrl.Result{}, nil
```

Comment:
This only changes the behaviour for RayJobs in terminal state. To be on the safe side, this keeps the requeueing / polling logic for RayJobs in "RayCluster deployment" phase, as well as for RayJobs in running state.  

Generally, it's a desired property from the performance / scalability standpoint to have controllers converging into a stationary state, that reacts to events, rather than entering an infinite loop, with a polling rate as small as `RayJobDefaultRequeueDuration`, which defaults to 3s.

For terminal states, I think it's safe to assume that no events are going to be missed, otherwise it's a bug that can be fixed. And to mitigate any discrepancy, the `SyncPeriod` manager option could be changed from the default, 10 hours, to something of an order of magnitude lower, even down to a few 10s of minutes.